src/middleware/with-graphql-query.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Inject the given graphql query into your request body.
+ *
+ * Your request can optionally specify the query variables you want:
+ * > client.Graphql.getResources({ variables: { id: 100 } })
+ *
+ */
+const WithGraphqlQuery = query => () => ({
+  prepareRequest (next) {
+    return next().then(request => {
+      if (request.requestParams) {
+        const variables = { ...request.requestParams }
+
+        const newRequest = request.enhance({
+          body: { ...variables, query }
+        })
+
+        Object.keys(variables).forEach(
```

Comment:
I had to delete whatever I query with, otherwise Mappersmith will try to add it as query params e.g. `/?variables=...`. Happy to hear if this can be done in a different way.